backend/s3/s3.go

```diff
@@ -2178,6 +2213,18 @@ func (f *Fs) makeBucket(ctx context.Context, bucket string) error {
 // Returns an error if it isn't empty
 func (f *Fs) Rmdir(ctx context.Context, dir string) error {
 	bucket, directory := f.split(dir)
+	// Remove directory marker file
+	if f.opt.DirectoryMarkers && bucket != "" && directory != "" {
+		markerFilePath := fmt.Sprintf("%s/", directory)
+		if f.rootDirectory != "" {
+			markerFilePath = fmt.Sprintf("%s/%s", f.rootDirectory, markerFilePath)
+		}
+		markerFileObject := &Object{
+			fs:     f,
+			remote: markerFilePath,
+		}
+		markerFileObject.Remove(ctx)
```

Comment:
[linter wants you to](https://github.com/rclone/rclone/runs/3263796986?check_suite_focus=true#step:14:23)
```
_ = markerFileObject.Remove(ctx)
```